### ISSUE TITLE
Computer science questions!!

### ISSUE THREAD

**Issue by srfoster** [thoughtstem/morugamu#25]
I'm collecting a list of computer science questions that I'll try to organize into a presentation.

Anything goes, but here are some examples:

* Why are there so many programming languages?
* What is Java for?
* What is functional programming?
* Is super-intelligent AI going to take over the world?
* Should I major in Computer Science?
* Should I minor in Computer Science?
* Who is the best programmer in the world?
* How do I get a job at Google?

**Comment by jlaxamana1298** [comment]
If I wanted to start learning and experimenting with machine learning and machine learning algorithms, where is the best place to start?

**Comment by gbm2cool** [comment]
What are ways to practice and improve my programming skills when I'm not doing this internship? (Online lessons, personal projects, etc?)

**Comment by Jordanf2018** [comment]
How is computer science used in careers?
What is functional programming?
What exactly makes computer science a science and not a practice or art?
How would computer science impact the development of machinery, vehicles, and robotics?
Is there a ONE language every single CS student should know?
What are the differences in the languages?
What makes CS an enticing minor/major?(Or why did you choose CS?)

**Comment by ctran037** [comment]
How should I take the experiences I learned through this internship and apply them future job interviews?

**Comment by JuliaB1** [comment]
How does networking and other back-end, server-type stuff work?

(I realize this is a pretty vague and open-ended question, so to kinda narrow my question a bit: I thought the QR-logging in system that ThoughtSTEM has implemented for the camps was pretty cool, and I was curious about how that worked)

**Comment by Soulst** [comment]
For a CS student or anyone seeking a career in CS/programming, what kinds of projects or portfolio really interests employers? Any tips?

I am often told that I should be working on projects of my own to gather a portfolio. However, I worry that employers won't care about some silly little game I make. Should I stick to whatever interests me purely? Or strive to impress THAT particular company with something more relevant.

**Comment by simon0302** [comment]
1. What computer science knowledge would I need if I want to specialize in controls?
2. With regard to the long run, should I minor or double major in computer science?
3. What is functional programming?
4. How do I start learning more about machine learning?
5. If society is so concerned with the possibility of a singularity where AI will be able to overtake humans, why are computer scientists still pursuing AI?
6. Should I take classes on UI?

**Comment by taco-coder** [comment]
Obviously, at the moment there is a huge demand for computer scientists and programmers. In addition to that though, there is an increasing number of people studying computer science. How do you think the current demand of the industry will evolve in the next 15 or so years? Will there still be a high demand or will CompSci become like mechanical engineering where the industry becomes saturated and demand dies down?

**Comment by chijj** [comment]
Would it be a good decision to switch to Computer Science even though I am struggling with this project?

**Comment by taco-coder** [comment]
In the job market, you're competing against a thousand other coders who know the same languages and information as you. What nontechnical skill/trait(s) do you believe give someone the edge over the competition?

**Comment by alan-lam** [comment]
Do you think functional programming will ever be used in industry and, if so, how would that happen?

**Comment by katsteve** [comment]
What sorts of minors would best compliment a major in computer science?
Besides the traditional career of a programmer, where else would a major in computer science take you?

**Comment by mollycompton** [comment]
Is it better to learn multiple programming languages at once or one at a time?
What do you do if the languages you know aren't popular anymore?
How do you decide which languages to learn?
Should everyone learn about cybersecurity or is that just a special topic?

**Comment by hyphy22** [comment]
What direction in computer languages/API's/databases should I take in order to prepare myself to become a pen(penetration) tester?
What will help me stand out amongst other computer science majors looking for jobs as well?

**Comment by Meleleyva** [comment]
How can you inspire yourself to become more creative and to keep that going?

**Comment by JavaCafe01** [comment]
How much of a language should you know before pursuing a career for it?

**Comment by katsteve** [comment]
How was ThoughtSTEM funded? What is the origin story of ThoughtSTEM?

**Comment by alan-lam** [comment]
Is getting a CS degree with the intention of becoming a web developer a waste of a degree? (Since web development seems to be mostly a self-taught subject)

With the vast amount of (free) resources available for learning different areas of CS, would it be a waste of time, money, and money - or just not as worth it - to go for a CS degree? (Disregarding the degree requirement for some jobs as that can be (arguably) overlooked if someone has a ton of experience/skills)

**Comment by taco-coder** [comment]
Will CodeSpells ever be added to the ThoughtSTEM curriculum? It's a cool idea and I think it'd be interesting for those more serious about game development (older than the Minecraft modders) .

**Comment by alan-lam** [comment]
With the curriculum for the coding camps, what do you want the students to walk away with? Or maybe not walk away with? What were the challenges of coming up with the curriculum?

From Soulst: "For a CS student or anyone seeking a career in CS/programming, what kinds of projects or portfolio really interests employers? Any tips?

I am often told that I should be working on projects of my own to gather a portfolio. However, I worry that employers won't care about some silly little game I make. Should I stick to whatever interests me purely? Or strive to impress THAT particular company with something more relevant."
